etc/picongpu/taurus-tud/knl_picongpu.profile.example

```diff
@@ -0,0 +1,42 @@
+# Text Editor for Tools #######################################################
+#   - examples: "nano", "vim", "emacs -nw", "vi" or without terminal: "gedit"
+#export EDITOR="nano"
+
+# General modules #############################################################
+#
+module purge
+module load cmake/3.9.0
+module load git
+module load openmpi/2.1.0-intel
+
+# Compilers ###################################################################
+### ICC
+module load intel/2017.2.174
+# Boost needs to be build yourself for now!
+# These paths need to be adjusted.
+export LD_LIBRARY_PATH=<BOOST_INSTALLATION_DIR>:$LD_LIBRARY_PATH
+export LD_LIBRARY_PATH=<BOOST_INSTALLATION_DIR>/lib:$LD_LIBRARY_PATH
+export BOOST_ROOT=<BOOST_INSTALLATION_DIR>
+
+export PICSRC=$HOME/src/picongpu
+export PIC_BACKEND="comp2b:native"
```

Comment:
typo? `omp2b:native` without `c`

Reply:
you should even set `omp2b:knl` if icc understands that, since the user will compile on the head node

Reply:
* `SBATCH --mem` is per node:
  ```
  --mem=<MB>
              Specify  the  real memory required per node in MegaBytes.
  ```

* @typo: Yes, no idea how this could happen

* `omp2b:knl` does not work:
  ```
  icc test.cpp -march=knl
  icc: command line warning #10159: invalid argument for option '-m'
  ```

Reply:
hm, `icc --help` offers the following:
```
-mtune=<cpu>
          optimize for a specific <cpu>
            slm        - processors that support Intel(R) SSE4.2 and MOVBE
                         instructions
            core-avx2  - processors that support Intel(R) Advanced Vector
                         Extensions 2 (Intel(R) AVX2)
            core-avx-i - processors that support Intel(R) Advanced Vector
                         Extensions (Intel(R) AVX), including instructions in
                         Intel(R) Core 2(TM) processors in process technology
                         smaller than 32nm
            corei7-avx - processors that support Intel(R) Advanced Vector
                         Extensions (Intel(R) AVX)
            corei7     - processors that support Intel(R) SSE4 Efficient
                         Accelerated String and Text Processing instructions
            atom       - processors that support MOVBE instructions
            core2      - Intel(R) Core 2(TM) processor family
            pentium-m  - Intel(R) Pentium(R) M processors
            pentium4   - Intel(R) Pentium(R) 4 processors
            pentium3   - Intel(R) Pentium(R) III processors (Linux only)
-march=<cpu>
          generate code exclusively for a given <cpu>
            slm        - processors that support Intel(R) SSE4.2 and MOVBE
                         instructions
            core-avx2  - processors that support Intel(R) Advanced Vector
                         Extensions 2 (Intel(R) AVX2)
            core-avx-i - processors that support Intel(R) Advanced Vector
                         Extensions (Intel(R) AVX), including instructions in
                         Intel(R) Core 2(TM) processors in process technology
                         smaller than 32nm
            corei7-avx - processors that support Intel(R) Advanced Vector
                         Extensions (Intel(R) AVX)
            corei7     - processors that support Intel(R) SSE4 Efficient
                         Accelerated String and Text Processing instructions
            atom       - processors that support MOVBE instructions
            core2      - Intel(R) Core 2(TM) processor family
            pentium-m  - Intel(R) Pentium(R) M processors
            pentium4   - Intel(R) Pentium(R) 4 processors
            pentium3   - Intel(R) Pentium(R) III processors (Linux only)
```

Additionally, the now deprecated offloading shows:
```
-qoffload-arch=<arch>[:<visa>]
          specify the target architecture to use when offloading code
            arch - Specifies the target architecture. On Intel(R) 64
                   architecture targeting Intel(R) MIC Architecture, arch can
                   be mic or mic-avx512. On IA-32 and Intel(R) 64 architectures
                   targeting Intel(R) Graphics Technology, arch can be
                   haswell, broadwell, cherryview, skylake, or broxton.
            visa - Specifies the virtual ISA (vISA) version to be used. This
                   optional argument is only allowed on IA-32 and Intel(R) 64
                   architectures targeting Intel(R) Graphics Technology. You
                   can specify either visa3.1 or visa3.2.
```

so maybe `omp2b:mic-avx512`?

Reply:
No, this is not working:
```
$ icc test.cpp -xMIC-AVX512
icc: command line warning #10121: overriding '-msse4.2' with '-xMIC-AVX512'
…
$ icc test.cpp -arch=MIC-AVX512
icc: command line warning #10006: ignoring unknown option '-arch=MIC-AVX512'
```
Of course best way would be to give the icc the `-xARCH` flag instead of `-march` as described here https://github.com/ComputationalRadiationPhysics/picongpu/issues/2206

Reply:
hm, actuall its [`-xmic-avx512`](https://www.alcf.anl.gov/files/Intel%20Compiler%20Optimization%20and%20Building%20for%20KNL.pdf)

Reply:
```
$ icc --help
…
-x<code>  generate specialized code to run exclusively on processors
          indicated by <code> as described below
 …
            MIC-AVX512
                    May generate Intel(R) Advanced Vector Extensions 512
                    (Intel(R) AVX-512) Foundation instructions, Intel(R)
                    AVX-512 Conflict Detection instructions, Intel(R) AVX-512
                    Exponential and Reciprocal instructions, Intel(R) AVX-512
                    Prefetch instructions for Intel(R) processors, and the
                    instructions enabled with CORE-AVX2. Optimizes for Intel(R)
                    processors that support Intel(R) AVX-512 instructions.
…
```

Reply:
tl;dr: Both are working ;)

Reply:
Even `-xmIc-AvX512` is working XD